### PR TITLE
remove unnecessary doc string macros, re-sync documentation

### PR DESCRIPTION
**Pull request by jakebolewski**

### CODE REVIEW

File: doc/stdlib/linalg.rst [hunk 1]
```diff
@@ -149,7 +149,7 @@ Linear algebra functions in Julia are largely implemented by calling functions f
 
    .. Docstring generated from Julia source
 
-   Compute an ``LDLt`` factorization of a real symmetric tridiagonal matrix such that ``A = L*Diagonal(d)*L'`` where ``L`` is a unit lower triangular matrix and ``d`` is a vector. The main use of an ``LDLt`` factorization ``F = ldltfact(A)`` is to solve the linear system of equations ``Ax = b`` with ``F\b``\ .
+   Compute an ``LDLt`` factorization of a real symmetric tridiagonal matrix such that ``A = L*Diagonal(d)*L'`` where ``L`` is a unit lower triangular matrix and ``d`` is a vector. The main use of an ``LDLt`` factorization ``F = ldltfact(A)`` is to solve the linear system of equations ``Ax = b`` with ``F``\ .
```
Comment (tkelman) [inline doc/stdlib/linalg.rst]: is this a mistake in escaping, or a bug?

File: doc/stdlib/linalg.rst [hunk 2]
```diff
@@ -904,7 +904,7 @@ Linear algebra functions in Julia are largely implemented by calling functions f
 
    .. Docstring generated from Julia source
 
-   If ``A`` has no negative real eigenvalue, compute the principal matrix logarithm of ``A``, i.e. the unique matrix :math:`X` such that :math:`e^X = A` and :math:`-\pi < Im(\lambda) < \pi` for all the eigenvalues :math:`\lambda` of :math:`X`. If ``A`` has nonpositive eigenvalues, a nonprincipal matrix function is returned whenever possible.
+   If ``A`` has no negative real eigenvalue, compute the principal matrix logarithm of ``A``, i.e. the unique matrix :math:`X` such that :math:`e^X = A` and :math:`-pi < Im(lambda) < pi` for all the eigenvalues :math:`lambda` of :math:`X`. If ``A`` has nonpositive eigenvalues, a nonprincipal matrix function is returned whenever possible.
```
Comment (tkelman) [inline doc/stdlib/linalg.rst]: losing some latex escaping here and in the rest of the diffs below - I guess the doc macro is necessary for those?

Comment (jakebolewski) [inline doc/stdlib/linalg.rst]: hmm, seems like it.  Is there an issue for this?

**Comment by vtjnash** [comment]
`@doc` isn't trivially unnecessary -- it has the side-effect of disabling the standard julia string processing

**Comment by jakebolewski** [comment]
I would have assumed that the two are synonymous, but now we have two different ways to do the same thing with subtle differences that are not documented.  We really should just have one syntax for this stuff that everyone uses.

**Comment by JeffBezanson** [comment]
@jakebolewski is right. I'm almost inclined to require `doc""" """`, to avoid gotchas like  #12737.

**Comment by jakebolewski** [comment]
I'd support requiring the `doc` form.  I wished `doc` was just a keyword / first class syntax but I think I lost that argument over the summer.

**Comment by JeffBezanson** [comment]
I'm fine with special-casing `doc` string literals to have the magic syntax. It seems better than the alternative of applying magic syntax to *all* string literals.

**Comment by StefanKarpinski** [comment]
I think the ship has sailed on requiring the `doc` prefix on string literals. Going from allowing bare doc strings in 0.4 to requiring the `doc` prefix in 0.5 is not going to go over well. The main issue is using `$` for inline latex. I propose using double backticks instead, which is [what Scholarly Markdown uses](http://scholarlymarkdown.com/Scholarly-Markdown-Guide.html#math). This has the additional advantage of rendering in a reasonable fashion (as inline LaTeX code) in Markdown engines that don't understand inline LaTeX. The main disadvantage is that it departs from Jupyter's Markdown flavor.

**Comment by jakebolewski** [comment]
> Isn't going to go over well.
As opposed to subtly breaking all code that uses arrays?

**Comment by StefanKarpinski** [comment]
How is this breaking all code that uses arrays?

**Comment by jakebolewski** [comment]
Getting a good documentation system *that just works* is one of the most important things to get right.  I'm just advocating that on the whole if we have to make something backwards incompatible to fix corner cases then we should, given the magnitude of other proposed changes for 0.5.

**Comment by StefanKarpinski** [comment]
No one is arguing that we should have a broken documentation system. I think we should get rid of all of the `doc` prefix versions of this and *only* have the bare doc string version and work on getting all the subtle docstring parser bugs taken care of. I feel that we've now taken care of most of the docstring parsing bugs. What is this business about breaking all of array usages?

**Comment by JeffBezanson** [comment]
Having only the bare string version is also fine with me.

**Comment by KristofferC** [comment]
Arraymageddon.

**Comment by StefanKarpinski** [comment]
How is Arraymageddon relevant to this issue?

**Comment by tkelman** [comment]
That "ship has sailed" pronouncements are not necessarily accurate for things like this.

There were some comments on thr FRBNY announcement post from people who had never seen Julia that raw docstrings make it difficult to distinguish code from non-code. I'd be +1 on always needing `doc`.

**Comment by JeffBezanson** [comment]
> distinguish code from non-code

Syntax highlighting?

**Comment by tkelman** [comment]
Changing the latex syntax would probably be good anyway, since there could be interpolation cases where you'd want normal string rules to apply. For example auto inserting the signature which we don't have a way of doing yet.

**Comment by StefanKarpinski** [comment]
I agree that syntax highlighting is a good approach to addressing this – especially since we'll want to syntax highlight the markdown as markdown. There was also a complaint about multiline comments, which I'm much more sympathetic to since it's at the top of my regrettable features list.

**Comment by tkelman** [comment]
Having Julia code be illegible in editors that have not been configured to highlight julia isn't great. Syntax highlighting is a convenience, it shouldn't be mandatory to understand things.

**Comment by JeffBezanson** [comment]
But I don't see how `doc` solves that problem either. To me three quotes is enough of a visual cue to separate the doc string. Anyway I assume python has the same "problem"?

**Comment by StefanKarpinski** [comment]
Yeah, I don't see what the issue is – strings get highlighted as strings if you don't have support for highlighting them as code. Guiding language designed based on the most broken possible editor configuration doesn't seem like it goes anywhere good.

**Comment by StefanKarpinski** [comment]
The comments @tkelman is referring to in case people want to judge the apt level of concern:

https://news.ycombinator.com/item?id=10671581

I see the issue and I think it's a problem for basically any form of extensive inline documentation.

**Comment by tkelman** [comment]
I'm going to close this since I think #14378 makes it redundant. Delete the branch if you agree.

**Pull request closed without merging**